### PR TITLE
Expand native lib installation instructions in README

### PR DESCRIPTION
**Pull request by mfitz**
While installing GeNet on an EC2 instance I realised that the README instructions around the underlying native libraries could use some improvement.

The rendered README can be found [here](https://github.com/arup-group/genet/tree/expand-native-libs-installation-docs). [This section](https://github.com/arup-group/genet/tree/expand-native-libs-installation-docs#installation-as-a-python-package) is where most of the change is, especially [this subsection](https://github.com/arup-group/genet/tree/expand-native-libs-installation-docs#installing-the-native-dependencies).

I haven't added any instructions for Windows (or Anaconda), because I don't know what they are. That is a gap that we should fill in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 
 [![DOI](https://zenodo.org/badge/265256468.svg)](https://zenodo.org/badge/latestdoi/265256468)
 
+## Table of Contents
```
Comment (mfitz) [inline README.md:5]: Nice to have a clickable table of contents. I found [this online tool](https://ecotrust-canada.github.io/markdown-toc/) that autogenerates a TOC from markdown. You still need to tweak some things in the resultant table, but it does most of the work.

File: README.md [hunk 2]
```diff
@@ -65,52 +90,85 @@ inspect or change it and save it out to file. Check out the
...
+functionality, you do not need to install this solver.
+Methods default to [CBC](https://projects.coin-or.org/Cbc), an open source solver.
+Another good open source choice is [GLPK](https://www.gnu.org/software/glpk/).
+The solver you use needs to support MILP - mixed integer linear programming.
+
+#### Installing the native dependencies
+The commands for installing the necessary native libraries vary according to the operating system you are using, for
+example:
+
+| OS       | Commands |
```
Comment (mfitz) [inline README.md:119]: I dunno what it is Windows users do...

Comment (KasiaKoz) [reply README.md:121]: yeah, I don't know either, i think a lot of people use a linux subsystem

**Review by KasiaKoz — APPROVED** [review]
❇️ very nice ❇️

**Pull request merged**